### PR TITLE
Expose OpenTelemetry APIs via MP Telemetry

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Fixes: #210

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    
+    <artifactId>microprofile-telemetry-api</artifactId>
+    <version>2.0-SNAPSHOT</version>
+    <packaging>pom</packaging>
+    <name>MicroProfile Telemetry API</name>
+    <dependencies>
+        <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-sdk</artifactId>
+    </dependency>
```
Comment (Azquelt) [inline api/pom.xml:33]: Indentation is inconsistent

File: api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-sdk-extension-autoconfigure</artifactId>
+    </dependency>
```
Comment (Azquelt) [inline api/pom.xml:37]: Had to think for a moment about the scope here, but I think the way you've done it is correct. :+1:

The user should depend on _this_ artifact with scope `provided`, but this artifact's dependencies should be scope `compile`.

**Review by Azquelt — CHANGES_REQUESTED** [review]
No issue linked and no PR description.

I think having an empty API jar which depends on the relevant levels of opentelemetry makes sense, but I'd expect to see an issue proposing and discussing adding this artifact. It's possible that this was discussed and agreed in the community call, but then I'd expect it to be documented that this was the case.

File: api/src/main/java/resources/META-INF/LICENSE [hunk 3]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
+                           Version 2.0, January 2004
+                        http://www.apache.org/licenses/
```
Comment (Azquelt) [inline api/src/main/java/resources/META-INF/LICENSE:3]: This file is located in `src/main/java/resources` as if it's to be added to a jar, but the packaging type for this artifact is `pom`

**Comment by Azquelt** [comment]
Also need to test this to check it actually functions as we want.

I think the two use cases are:
- Depending on it as a `provided` dependency to provide the OTel APIs at the correct level
- Adding it as a dependency to the `org.eclipse.microprofile:microprofile` artifact.

File: api/pom.xml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.eclipse.microprofile.telemetry</groupId>
+        <artifactId>microprofile-telemetry-parent</artifactId>
+        <version>2.0-SNAPSHOT</version>
+    </parent>
+    
+    <artifactId>microprofile-telemetry-api</artifactId>
+    <version>2.0-SNAPSHOT</version>
```
Comment (Azquelt) [inline api/pom.xml:26]: `version` is not needed here since it will be inherited from the parent

**Review by brunobat — CHANGES_REQUESTED** [review]
User side API should be very limited to OpenTelemetry API, instrumentation annotations and semantic conventions.

File: api/pom.xml (hunk 2 above)
Comment (Azquelt) [reply api/pom.xml]: You're right, I'm used to seeing this because we use it in the TCK, but users shouldn't need it when writing an application.

**Review by brunobat — APPROVED** [review]
Looks goo to me now.

File: api/src/main/java/resources/META-INF/LICENSE (hunk 3 above)
Comment (Azquelt) [reply api/src/main/java/resources/META-INF/LICENSE:3]: This still needs removed

**Review by Azquelt — CHANGES_REQUESTED** [review]

**Review by Azquelt — APPROVED** [review]

**Pull request merged**